### ISSUE TITLE
Make semicolon optional for clauses inside lambda clauses

### ISSUE THREAD

**Issue by jonaprieto** [anoma/juvix#2051]
I was checking on a function like the one shown below, and I felt like I wanted to refactor it using only one clause. This process requires more editing than I expected.

```agda
    go : B -> BinaryTree A -> B;
    go acc leaf := acc;
    go acc (node l a r) := f a (go acc l) (go acc r);
```

```diff
  let
    go : B -> BinaryTree A -> B;
    go acc :=
      λ {
-        | leaf := acc;
+        | leaf := acc
-        | (node l a r) := f a (go acc l) (go acc r);
+        | (node l a r) := f a (go acc l) (go acc r)
      };
```

Initially, it was expected that only adding a pipe (|) to denote branches, and the obvious other symbols like := would be necessary while leaving the semicolon (;). However, it turns out that the semicolon must also be removed. I understand the resemblance with the datatype def pattern, but *I don't think we should force that thing here*. Of course, the formatter should remove the semicolon at a later stage, but if people want to end the clause with the semicolon that should be fine.

**Comment by lukaszcz** [comment]
This will become irrelevant after implementing #2189

**Comment by lukaszcz** [comment]
* Closed by #2243
